src/pages/RestaurantsPage/RestaurantsPage.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Fade } from "react-awesome-reveal";
+
+import "./RestaurantsPage.scss";
+import { RootState, AppDispatch } from "../../redux-toolkit/store";
+import { fetchRestaurantsPageData } from "../../redux-toolkit/thunks/restaurantsPageThunk";
+import { fetchHomePageData } from "../../redux-toolkit/thunks/homePageThunk";
+import Card from "../../components/Card/Card";
+import { CardProps } from "../../models/types";
+
+const RestaurantsPage = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const [activeFilter, setActiveFilter] = useState("All");
+
+  const { restaurants } = useSelector((state: RootState) => state.restaurantsPage);
+  const { popularRestaurants } = useSelector((state: RootState) => state.homePage);
+
+  useEffect(() => {
+    dispatch(fetchRestaurantsPageData());
+    dispatch(fetchHomePageData());
+    window.scrollTo(0, 0);
+  }, [dispatch]);
+
+  const handleFilterClick = (filter: string) => {
+    setActiveFilter(filter);
+  };
+
+  const filteredCards = () => {
```

Comment:
Make it generic func that recive a type of filter and by this filter taking the right restaurants